stringutils/sendreceivemam.go

```diff
@@ -0,0 +1,85 @@
+/*
+MIT License
+Copyright (c) 2017 Harry Boer
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+//Package stringutils is a utility to create Masked Authenticated Messages (MAM's)
+package stringutils
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/iotaledger/giota"
+)
+
+//ToMAMTrytes checks its validity and casts to giota.Trytes.
+func ToMAMTrytes(t string) (tr giota.Trytes) {
+
+	// Check if input is a string
+	if !IsString(t) {
+		fmt.Println("Input is not a string. Please provide a valid string.")
+	}
+
+	trytes := ""
+	TryteValues := "9ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+	for i := 0; i < len(t); i++ {
```

Comment:
len() returns number of bytes in the string, but then you use the runes in CharCodeAt(), thats wrong. If you want the runes here use:

for _, asciiValue := range t {

Reply:
Yes, that's the way js does it. I'll have a look at it again, although I'm not sure why it would be wrong. Security?

Reply:
I took away len() and changed it to ranging over the string; however, I believe that’s the correct way to do it. I found numerous solutions like this. See 
https://stackoverflow.com/questions/19231506/go-golang-access-string-as-character-value

In any case, if I (or you) find a better solution we’ll change it.